lib/Serialization/Serialization.cpp

```diff
@@ -579,10 +579,11 @@ void Serializer::writeHeader(const SerializationOptions &options) {
     ModuleName.emit(ScratchRecord, M->getName().str());
 
     SmallString<32> versionStringBuf;
+    auto version = M->getASTContext().LangOpts.EffectiveLanguageVersion;
     llvm::raw_svector_ostream versionString(versionStringBuf);
-    versionString << Version::getCurrentLanguageVersion();
+    versionString << version;
```

Comment:
Currently, only the short version string is used for diagnostics; actual mismatches are detected using the serialization's VERSION_MAJOR/VERSION_MINOR (itself designed for a stable serialization format that we haven't gotten to yet). The intent is that if the short diagnostic string has changed, we can offer a better diagnostic than just "the module is too old". The full string is mostly just so that we can run `strings` on a swiftmodule and see where it came from.